pkg/virt-api/rest/subresource_test.go

```diff
@@ -1735,70 +1735,85 @@ var _ = Describe("VirtualMachineInstance Subresources", func() {
-		})
+		table.DescribeTable("Should fail pausing", func(running bool, paused bool, pauseOptions *v1.PauseOptions) {
 
-		It("Should fail pausing a running but paused VMI", func() {
+			expectVMI(running, paused)
 
-			expectVMI(true, true)
+			bytesRepresentation, _ := json.Marshal(pauseOptions)
+			request.Request.Body = io.NopCloser(bytes.NewReader(bytesRepresentation))
 
 			app.PauseVMIRequestHandler(request, response)
 
 			ExpectStatusErrorWithCode(recorder, http.StatusConflict)
-		})
-
-		It("Should fail unpausing a running, not paused VMI", func() {
-
-			expectVMI(true, false)
+		},
+			table.Entry("a not running VMI", false, false, &v1.PauseOptions{}),
+			table.Entry("a not running VMI with dry-run option", false, false, &v1.PauseOptions{DryRun: []string{k8smetav1.DryRunAll}}),
 
-			app.UnpauseVMIRequestHandler(request, response)
+			table.Entry("a running but paused VMI", true, true, &v1.PauseOptions{}),
+			table.Entry("a running but paused VMI with dry-run option", true, true, &v1.PauseOptions{DryRun: []string{k8smetav1.DryRunAll}}),
+		)
 
-			ExpectStatusErrorWithCode(recorder, http.StatusConflict)
-		})
+		table.DescribeTable("Should fail unpausing", func(running bool, paused bool, unpauseOptions *v1.UnpauseOptions) {
 
-		It("Should fail unpausing a not running VMI", func() {
+			expectVMI(running, paused)
 
-			expectVMI(false, false)
+			bytesRepresentation, _ := json.Marshal(unpauseOptions)
+			request.Request.Body = io.NopCloser(bytes.NewReader(bytesRepresentation))
 
 			app.UnpauseVMIRequestHandler(request, response)
 
 			ExpectStatusErrorWithCode(recorder, http.StatusConflict)
-		})
+		},
+			table.Entry("a running, not paused VMI", true, false, &v1.UnpauseOptions{}),
+			table.Entry("a running, not paused VMI with dry-run option", true, false, &v1.UnpauseOptions{DryRun: []string{k8smetav1.DryRunAll}}),
+
+			table.Entry("a not running VMI", false, false, &v1.UnpauseOptions{}),
+			table.Entry("a not running VMI with dry-run option", false, false, &v1.UnpauseOptions{DryRun: []string{k8smetav1.DryRunAll}}),
+		)
+
+		table.DescribeTable("Should unpause a running, paused VMI", func(running bool, paused bool, unpauseOptions *v1.UnpauseOptions) {
 
-		It("Should unpause a running, paused VMI", func() {
 			backend.AppendHandlers(
 				ghttp.CombineHandlers(
 					ghttp.VerifyRequest("PUT", "/v1/namespaces/default/virtualmachineinstances/testvmi/unpause"),
 					ghttp.RespondWith(http.StatusOK, ""),
 				),
 			)
-			expectVMI(true, true)
+			expectVMI(running, paused)
+
+			bytesRepresentation, _ := json.Marshal(unpauseOptions)
+			request.Request.Body = io.NopCloser(bytes.NewReader(bytesRepresentation))
 
 			app.UnpauseVMIRequestHandler(request, response)
 
 			Expect(response.StatusCode()).To(Equal(http.StatusOK))
-		})
+
+		},
+			table.Entry("", true, true, &v1.UnpauseOptions{}),
+			table.Entry("with dry-run option", true, true, &v1.UnpauseOptions{DryRun: []string{k8smetav1.DryRunAll}}),
```

Comment:
The resulting sentence for this test will be "Should unpause a running, paused VMI with dry-run option".
That sounds incorrect, nothing should actually happen in dry mode...

Reply:
Thanks for the review and suggestions. Fixed

Reply:
Same as above, it doesn't seem needed to add the booleans as parameters.

In general, having random booleans around is hard to read and usually frowned upon.
How would you feel about defining 4 consts at the top of the file like "Paused=true, UnPaused=false, Running=true, NotRunning=false"?
That way we could write calls like `expectVMI(Running, Paused)`, without having to create function parameters for them!

Reply:
I like it! It will increase readability. Thanks so much!